src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,173 @@
+package duke;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Parses all input commands given by the user.
+ */
+public class Parser {
+
+    //The DukeList to store all tasks given to the Duke bot.
+    private DukeList list;
+
+    //The timeParser to parse and format all times given.
+    private static DateAndTime timeParser = new DateAndTime();
+
+
+    /**
+     * Instantiates a new Parser class
+     * @param list the DukeList that contains all tasks at hand.
+     */
+    public Parser(DukeList list) {
+        this.list = list;
+    }
+
+    /**
+     * Helps to parse and manage the user's inputs. "list" shows the current list, "mark / unmark"
+     * helps to mark or unmark the specific task in the list, and todo/deadline/event
+     * adds todo/deadline/event tasks respectively.
+     *
+     * @param tally takes in the input string.
+     */
+    public void messageHandler(String tally) {
```

Comment:
I think method names should be verbs, instead of nouns? Maybe this could be renamed to "handleMessage"